genesyscloud/architect_flow/resource_genesyscloud_architect_flow_proxy.go

```diff
@@ -83,46 +98,143 @@ func (a *architectFlowProxy) GetAllFlows(ctx context.Context, name string, varTy
+	return a.createExportJobAttr(a, flowId)
+}
+
+// getExportJobStatusById retrieves the status of an architect flow export job.
+// Implementation function: getExportJobStatusByIdFn
+// Parameters:
+//   - jobId: string - The ID of the export job to check
+//
+// Returns:
+//   - *platformclientv2.ArchitectFlowJobStatusResponse: The job status response
+//   - *platformclientv2.APIResponse: The raw API response
+//   - error: An error if the job status cannot be retrieved or if the response is invalid
+//
+// The function will return an error if the job status or status field is nil in the response.
+func (a *architectFlowProxy) getExportJobStatusById(jobId string) (*platformclientv2.Architectexportjobstateresponse, *platformclientv2.APIResponse, error) {
+	return a.getExportJobStatusByIdAttr(a, jobId)
+}
+
+// pollExportJobForDownloadUrl polls an architect flow export job until it completes or times out.
+// It checks the job status periodically and waits for the download URL to become available.
+// Implementation function: pollExportJobForDownloadUrlFn
+//
+// Parameters:
+//   - a: *architectFlowProxy - The architect flow proxy instance
+//   - jobId: string - The ID of the export job to poll
+//
+// Returns:
+//   - string: The download URL for the exported flow
+//   - error: An error if the polling times out or if there's an issue getting the job status
+//
+// The function will timeout after 90 seconds if the job doesn't complete.
+// It polls the job status every 1 second.
+func (a *architectFlowProxy) pollExportJobForDownloadUrl(jobId string) (downloadUrl string, err error) {
+	return a.pollExportJobForDownloadUrlAttr(a, jobId)
+}
+
 func getFlowIdByNameAndTypeFn(ctx context.Context, a *architectFlowProxy, name, varType string) (string, *platformclientv2.APIResponse, bool, error) {
 	var (
-		matchedFlowIds []string
-		typeDetails    string
-		types          []string
+		matchedFlows    []platformclientv2.Flow
+		types           []string
+		noFlowsFoundErr = fmt.Errorf("no flows found with name '%s' and type '%s'", name, varType)
 	)
 
 	if varType != "" {
 		types = append(types, varType)
 	}
 
-	if varType != "" {
-		typeDetails = fmt.Sprintf("type '%s'", varType)
-	}
-	noFlowsFoundErr := fmt.Errorf("no flows found with name '%s' %s", name, typeDetails)
-
 	flows, resp, err := a.GetAllFlows(ctx, name, types)
 	if err != nil {
 		return "", resp, false, err
 	}
 
+	// no flows found with that name
 	if flows == nil || len(*flows) == 0 {
 		return "", nil, true, noFlowsFoundErr
 	}
 
 	for _, flow := range *flows {
 		if *flow.Name == name {
-			matchedFlowIds = append(matchedFlowIds, *flow.Id)
+			matchedFlows = append(matchedFlows, flow)
 		}
-		if len(matchedFlowIds) > 1 {
-			return "", nil, true, fmt.Errorf("found more than one flow that matched the name '%s'", name)
+	}
+
+	// if more than one flow is found, return the one that matches the type also
```

Comment:
Shouldn't we also ensure the VarType _always_ matches, and not just if there is more than one?

Reply:
![Uploading Screenshot 2025-03-24 at 14.25.32.png…]()

Reply:
```bash
╷
│ Warning: 'type' should be provided to the genesyscloud_flow data source
│ 
│   with data.genesyscloud_flow.flow,
│   on main.tf line 1, in data "genesyscloud_flow" "flow":
│    1: data "genesyscloud_flow" "flow" {
│ 
│ Please provide a value for the field 'type' in the genesyscloud_flow data source as it will become required in a later version.
╵

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

flow_id = "0387e182-5c0c-44c2-9f34-66c168ebda3e"
```